components/net/http_loader.rs

```diff
@@ -1637,40 +1588,108 @@ async fn http_network_or_cache_fetch(
         .await;
     }
 
-    // Step 11
+    // Step 15. If response’s status is 407, then:
     if let Some((StatusCode::PROXY_AUTHENTICATION_REQUIRED, _)) = response.status.as_ref() {
-        // Step 1
+        // Step 15.1 If request’s window is "no-window", then return a network error.
         if request_has_no_window {
             return Response::network_error(NetworkError::Internal(
                 "Can't find Window object".into(),
             ));
         }
 
-        // Step 2
-        // TODO: Spec says requires testing on Proxy-Authenticate headers
+        // (Step 15.2 does not exist, requires testing on Proxy-Authenticate headers)
 
-        // Step 3
-        // FIXME: Prompt the user for proxy authentication credentials
+        // FIXME: Step 15.3 If fetchParams is canceled, then return the appropriate network error for fetchParams.
+        // FIXME: Step 15.4 Prompt the end user as appropriate in request’s window and store the
+        //                  result as a proxy-authentication entry.
+
+        // Step 15.5 Set response to the result of running HTTP-network-or-cache fetch given fetchParams.
 
         // Wrong, but will have to do until we are able to prompt the user
         // otherwise this creates an infinite loop
         // We basically pretend that the user declined to enter credentials
         return response;
-
-        // Step 4
-        // return http_network_or_cache_fetch(request, authentication_fetch_flag,
-        //                                    cors_flag, done_chan, context);
     }
 
-    // Step 12
+    // FIXME: Step 16. If all of the following are true:
+    //                 * response’s status is 421
+    //                 * isNewConnectionFetch is false
+    //                 * request’s body is null, or request’s body is non-null and request’s body’s
+    //                   source is non-null
+    //                 then: [..]
+
+    // Step 17. If isAuthenticationFetch is true, then create an authentication entry for request and the given realm.
     if authentication_fetch_flag {
-        // TODO Create the authentication entry for request and the given realm
+        // TODO
     }
 
-    // Step 13
+    // Step 18. Return response.
     response
 }
 
+/// <https://fetch.spec.whatwg.org/#cross-origin-resource-policy-check>
+///
+/// This is obtained from [cross_origin_resource_policy_check]
+#[derive(PartialEq)]
+enum CrossOriginResourcePolicy {
+    Allowed,
+    Blocked,
+}
+
+// TODO: Judging from the name, this appears to be https://fetch.spec.whatwg.org/#cross-origin-resource-policy-check,
+//       but the steps aren't even close to the spec. Perhaps this needs to be rewritten?
```

Comment:
Yes very good TODO. I think it would be great to have an issue for all those TODOs, with checkboxes.

Reply:
I opened #33615 for this (seperate issue since its not directly related to the http network or cache fetch)